### ISSUE TITLE
Matlab Controllers Broken

### ISSUE THREAD

**Issue by DavidMansolino** [omichel/webots#63]
**Describe the Bug**
Matlab controllers are broken (this might be recent as it was working ~1month ago).

**Steps to Reproduce**
1. Open the matlab sample and run it.

**Screenshots**
```
MATLAB crash file:C:\Users\David\AppData\Local\Temp\matlab_crash_dump.8892-1:


--------------------------------------------------------------------------------
             Assertion detected at mer. déc. 12 04:09:53 2018 +0100
--------------------------------------------------------------------------------

Configuration:
  Crash Decoding           : Disabled - No sandbox or build area path
  Crash Mode               : continue (default)
  Default Encoding         : windows-1252
  Deployed                 : false
  Graphics Driver          : Unknown hardware 
  Graphics card 1          : NVIDIA ( 0x10de ) NVIDIA GeForce GTX 970 Version 25.21.14.1694 (2018-11-12)
  Java Crash Report        : C:\Users\David\AppData\Local\Temp\hs_error_pid8892.log
  Java Version             : Java 1.8.0_144-b01 with Oracle Corporation Java HotSpot(TM) 64-Bit Server VM mixed mode
  MATLAB Architecture      : win64
  MATLAB Entitlement ID    : 1376434
  MATLAB Root              : C:\Program Files\MATLAB\R2018a
  MATLAB Version           : 9.4.0.813654 (R2018a)
  OpenGL                   : hardware
  Operating System         : Microsoft Windows 10 Famille
  Process ID               : 8892
  Processor ID             : x86 Family 6 Model 60 Stepping 3, GenuineIntel
  Session Key              : 6e2697c1-508e-4e34-8cbd-6994f1ad538c
  Window System            : Version 10.0 (Build 17134)

Fault Count: 1


Assertion in void __cdecl `anonymous-namespace'::mwJavaAbort(void) at b:\matlab\src\jmi\jmi\javainit.cpp line 1461

Register State (captured):
  RAX = 00000000fea22ef8  RBX = 0000000000000000
  RCX = 00000000043ebed0  RDX = 0000000000000000
  RSP = 00000000043ebe50  RBP = 0000000021218858
  RSI = 00000000fea22ef8  RDI = 00000000fea22ef8
 
   R8 = 00008d9ae3973cff   R9 = 00007ffd17950000
  R10 = 00000000fe95a298  R11 = 00000000fea12eec
  R12 = 00000000043ec510  R13 = 00000000fea12e28
  R14 = 00000000fea22e78  R15 = 0000000021218858
 
  RIP = 00000000fe81c44a  EFL = 00000206
 
   CS = 0033   FS = 0053   GS = 002b

Stack Trace (captured):
[  0] 0x00000000fe816063                              bin\win64\libmwfl.dll+00155747
[  1] 0x00000000fe814658                              bin\win64\libmwfl.dll+00149080
[  2] 0x00000000fe817f68                              bin\win64\libmwfl.dll+00163688
[  3] 0x00000000fe81780e                              bin\win64\libmwfl.dll+00161806
[  4] 0x00000000fe81bdc8                              bin\win64\libmwfl.dll+00179656
[  5] 0x00000000211abcec                                  bin\win64\jmi.dll+00703724
[  6] 0x0000000059b36d1d C:\Program Files\MATLAB\R2018a\sys\java\jre\win64\jre\bin\server\jvm.dll+02714909
[  7] 0x0000000059b1ef3b C:\Program Files\MATLAB\R2018a\sys\java\jre\win64\jre\bin\server\jvm.dll+02617147
[  8] 0x0000000059b37a46 C:\Program Files\MATLAB\R2018a\sys\java\jre\win64\jre\bin\server\jvm.dll+02718278
[  9] 0x0000000059b3c33c C:\Program Files\MATLAB\R2018a\sys\java\jre\win64\jre\bin\server\jvm.dll+02736956
[ 10] 0x0000000059bd2938 C:\Program Files\MATLAB\R2018a\sys\java\jre\win64\jre\bin\server\jvm.dll+03352888
[ 11] 0x00007ffd1ed4ed3d                      C:\WINDOWS\SYSTEM32\ntdll.dll+00650557
[ 12] 0x00007ffd1ecb6c86                      C:\WINDOWS\SYSTEM32\ntdll.dll+00027782
[ 13] 0x00007ffd1ed4dc6e                      C:\WINDOWS\SYSTEM32\ntdll.dll+00646254
[ 14] 0x000000018005e8fe C:\Program Files\Webots\msys64\usr\bin\msys-2.0.dll+00125182
[ 15] 0x000000018005fea8 C:\Program Files\Webots\msys64\usr\bin\msys-2.0.dll+00130728
[ 16] 0x00007ffd1ed156b8                      C:\WINDOWS\SYSTEM32\ntdll.dll+00415416
[ 17] 0x00007ffd1ecb722c                      C:\WINDOWS\SYSTEM32\ntdll.dll+00029228
[ 18] 0x00007ffd1ed4dc6e                      C:\WINDOWS\SYSTEM32\ntdll.dll+00646254
[ 19] 0x000000008044344c                                   <unknown-module>+00000000
```

**System**
 - Operating System: Windows 10
 - Matlab Version: R2018a

**Comment by DavidMansolino** [comment]
I am actually not so sure, as I can now reproduce this problem even with a clean environment.
But I have been able to better isolate the problem:
  - The crash happens only when the 'plot' (or similar) function is called.
  - If I manually launch Matlab (with same arguments) outside of Webots from the msys console it is working fine.

**Comment by DavidMansolino** [comment]
I am having another stack with an even simpler controller in another simulation.

### Controller
```
timestep = wb_robot_get_basic_time_step();

while wb_robot_step(TIME_STEP) ~= -1  % main loop.

end
```

### Stack
```
MATLAB crash file:C:\Users\David\AppData\Local\Temp\matlab_crash_dump.29020-1:


--------------------------------------------------------------------------------
         Access violation detected at lun. janv. 14 04:48:38 2019 +0100
--------------------------------------------------------------------------------

Configuration:
  Crash Decoding           : Disabled - No sandbox or build area path
  Crash Mode               : continue (default)
  Default Encoding         : windows-1252
  Deployed                 : false
  Graphics Driver          : Unknown hardware 
  Graphics card 1          : NVIDIA ( 0x10de ) NVIDIA GeForce GTX 970 Version 25.21.14.1694 (2018-11-12)
  Java Version             : Java 1.8.0_144-b01 with Oracle Corporation Java HotSpot(TM) 64-Bit Server VM mixed mode
  MATLAB Architecture      : win64
  MATLAB Entitlement ID    : 1376434
  MATLAB Root              : C:\Program Files\MATLAB\R2018a
  MATLAB Version           : 9.4.0.813654 (R2018a)
  OpenGL                   : hardware
  Operating System         : Microsoft Windows 10 Famille
  Process ID               : 29020
  Processor ID             : x86 Family 6 Model 60 Stepping 3, GenuineIntel
  Session Key              : 430db01c-731e-4875-b13e-db5708e486a5
  Window System            : Version 10.0 (Build 17134)

Fault Count: 1


Abnormal termination

Register State (from fault):
  RAX = 0000000000000000  RBX = 00000000ed5535d0
  RCX = 00000000043fdd20  RDX = 00000000ed5535b0
  RSP = 00000000043e7eb0  RBP = 0000000069989630
  RSI = 0000000069989770  RDI = 00000000ed5535b0
 
   R8 = 0000000000000000   R9 = 00000000ed5535b0
  R10 = 00000000e6e1f020  R11 = 0000000000000000
  R12 = 00000000ed55b348  R13 = 00000000043ece50
  R14 = 0000000000000000  R15 = 00000000225b5630
 
  RIP = 00000000170453de  EFL = 00010246
 
   CS = 0033   FS = 0053   GS = 002b

Stack Trace (from fault):
[  0] 0x00000000170453de                            bin\win64\pgo\m_lxe.dll+00414686
[  1] 0x0000000017045378                            bin\win64\pgo\m_lxe.dll+00414584
[  2] 0x000000001704657a                            bin\win64\pgo\m_lxe.dll+00419194
[  3] 0x000000001701e0e0                            bin\win64\pgo\m_lxe.dll+00254176
[  4] 0x00000000170215fc                            bin\win64\pgo\m_lxe.dll+00267772
[  5] 0x00000000163f7a76                    bin\win64\pgo\m_interpreter.dll+00096886 inRegisterMFunctionFromOpaqueArray+00019986
[  6] 0x00007ff9982bc220 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00049696 _C_specific_handler+00000560
[  7] 0x00007ff9982b29b2 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00010674 _FrameUnwindFilter+00001074
[  8] 0x00007ff9a2aeed63                      C:\WINDOWS\SYSTEM32\ntdll.dll+00650595 RtlCaptureContext+00000995
[  9] 0x00000000163f7211                    bin\win64\pgo\m_interpreter.dll+00094737 inRegisterMFunctionFromOpaqueArray+00017837
[ 10] 0x00007ff9982bc220 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00049696 _C_specific_handler+00000560
[ 11] 0x00007ff9982b29b2 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00010674 _FrameUnwindFilter+00001074
[ 12] 0x00007ff9a2aeed63                      C:\WINDOWS\SYSTEM32\ntdll.dll+00650595 RtlCaptureContext+00000995
[ 13] 0x00000000163e5d7c                    bin\win64\pgo\m_interpreter.dll+00023932 MathWorks::MException::CreateMException+00000076
[ 14] 0x00000000186428f1                            bin\win64\pgo\m_lxe.dll+23472369 mwboost::serialization::singleton<mwboost::archive::detail::archive_serializer_map<mwboost::archive::binaryTerm_oarchive> >::get_instance+21036069
[ 15] 0x00007ff9982bc220 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00049696 _C_specific_handler+00000560
[ 16] 0x00007ff9982b29b2 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00010674 _FrameUnwindFilter+00001074
[ 17] 0x00007ff9a2aeed63                      C:\WINDOWS\SYSTEM32\ntdll.dll+00650595 RtlCaptureContext+00000995
[ 18] 0x00000000171d0016                            bin\win64\pgo\m_lxe.dll+02031638 mwboost::serialization::singleton<mwboost::archive::detail::pointer_oserializer<mwboost::archive::binary_oarchive,MathWorks::lxe::PreLineExecutionEvent> >::get_instance+00337014
[ 19] 0x0000000018643fa0                            bin\win64\pgo\m_lxe.dll+23478176 mwboost::serialization::singleton<mwboost::archive::detail::archive_serializer_map<mwboost::archive::binaryTerm_oarchive> >::get_instance+21041876
[ 20] 0x00007ff9982bc220 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00049696 _C_specific_handler+00000560
[ 21] 0x00007ff9982b29b2 C:\Program Files\MATLAB\R2018a\bin\win64\VCRUNTIME140.dll+00010674 _FrameUnwindFilter+00001074
[ 22] 0x00007ff9a2aeed63                      C:\WINDOWS\SYSTEM32\ntdll.dll+00650595 RtlCaptureContext+00000995
[ 23] 0x00000000171dd4e3                            bin\win64\pgo\m_lxe.dll+02086115 mwboost::serialization::singleton<mwboost::archive::detail::pointer_oserializer<mwboost::archive::binary_oarchive,MathWorks::lxe::PreLineExecutionEvent> >::get_instance+00391491
[ 24] 0x00000000171dd45b                            bin\win64\pgo\m_lxe.dll+02085979 mwboost::serialization::singleton<mwboost::archive::detail::pointer_oserializer<mwboost::archive::binary_oarchive,MathWorks::lxe::PreLineExecutionEvent> >::get_instance+00391355
[ 25] 0x000000001704f30d                            bin\win64\pgo\m_lxe.dll+00455437
[ 26] 0x000000001704b335                            bin\win64\pgo\m_lxe.dll+00439093
[ 27] 0x000000001704a82e                            bin\win64\pgo\m_lxe.dll+00436270
[ 28] 0x0000000017048b35                            bin\win64\pgo\m_lxe.dll+00428853
[ 29] 0x0000000017048e9e                            bin\win64\pgo\m_lxe.dll+00429726
[ 30] 0x0000000017048e29                            bin\win64\pgo\m_lxe.dll+00429609
[ 31] 0x000000001621c51a                     bin\win64\pgo\m_dispatcher.dll+00050458 Mfh_file::dispatch_fh_impl+00001114
[ 32] 0x000000001621c0b2                     bin\win64\pgo\m_dispatcher.dll+00049330 Mfh_file::dispatch_fh+00000062
[ 33] 0x0000000016214674                     bin\win64\pgo\m_dispatcher.dll+00018036 Mfunction_handle::dispatch+00000420
[ 34] 0x0000000017042aa1                            bin\win64\pgo\m_lxe.dll+00404129
[ 35] 0x00000000170c8e25                            bin\win64\pgo\m_lxe.dll+00953893 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00115205
[ 36] 0x00000000170ca808                            bin\win64\pgo\m_lxe.dll+00960520 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00121832
[ 37] 0x00000000170cb403                            bin\win64\pgo\m_lxe.dll+00963587 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00124899
[ 38] 0x00000000170cc836                            bin\win64\pgo\m_lxe.dll+00968758 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130070
[ 39] 0x00000000170cd3f0                            bin\win64\pgo\m_lxe.dll+00971760 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00133072
[ 40] 0x00000000170cc97f                            bin\win64\pgo\m_lxe.dll+00969087 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130399
[ 41] 0x00000000170cca7e                            bin\win64\pgo\m_lxe.dll+00969342 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130654
[ 42] 0x000000001704e8a5                            bin\win64\pgo\m_lxe.dll+00452773
[ 43] 0x000000001704b335                            bin\win64\pgo\m_lxe.dll+00439093
[ 44] 0x000000001704a82e                            bin\win64\pgo\m_lxe.dll+00436270
[ 45] 0x0000000017051b9a                            bin\win64\pgo\m_lxe.dll+00465818
[ 46] 0x000000001705133d                            bin\win64\pgo\m_lxe.dll+00463677
[ 47] 0x000000001705144e                            bin\win64\pgo\m_lxe.dll+00463950
[ 48] 0x000000001702033f                            bin\win64\pgo\m_lxe.dll+00262975
[ 49] 0x00000000170205c4                            bin\win64\pgo\m_lxe.dll+00263620
[ 50] 0x00000000163e5e19                    bin\win64\pgo\m_interpreter.dll+00024089 inFullEvalFcn+00000105
[ 51] 0x000000001621a9f4                     bin\win64\pgo\m_dispatcher.dll+00043508 mdFindFunctionFromUserSpecifiedName+00012500
[ 52] 0x0000000016214610                     bin\win64\pgo\m_dispatcher.dll+00017936 Mfunction_handle::dispatch+00000320
[ 53] 0x0000000017042aa1                            bin\win64\pgo\m_lxe.dll+00404129
[ 54] 0x00000000170c8e25                            bin\win64\pgo\m_lxe.dll+00953893 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00115205
[ 55] 0x00000000170ca808                            bin\win64\pgo\m_lxe.dll+00960520 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00121832
[ 56] 0x00000000170cb403                            bin\win64\pgo\m_lxe.dll+00963587 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00124899
[ 57] 0x00000000170cc836                            bin\win64\pgo\m_lxe.dll+00968758 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130070
[ 58] 0x00000000170cd3f0                            bin\win64\pgo\m_lxe.dll+00971760 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00133072
[ 59] 0x00000000170cc97f                            bin\win64\pgo\m_lxe.dll+00969087 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130399
[ 60] 0x00000000170cca7e                            bin\win64\pgo\m_lxe.dll+00969342 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130654
[ 61] 0x000000001704e8a5                            bin\win64\pgo\m_lxe.dll+00452773
[ 62] 0x000000001704b335                            bin\win64\pgo\m_lxe.dll+00439093
[ 63] 0x000000001704a82e                            bin\win64\pgo\m_lxe.dll+00436270
[ 64] 0x0000000017048b35                            bin\win64\pgo\m_lxe.dll+00428853
[ 65] 0x0000000017048e9e                            bin\win64\pgo\m_lxe.dll+00429726
[ 66] 0x0000000017048e29                            bin\win64\pgo\m_lxe.dll+00429609
[ 67] 0x000000001621c51a                     bin\win64\pgo\m_dispatcher.dll+00050458 Mfh_file::dispatch_fh_impl+00001114
[ 68] 0x000000001621c0b2                     bin\win64\pgo\m_dispatcher.dll+00049330 Mfh_file::dispatch_fh+00000062
[ 69] 0x0000000016214674                     bin\win64\pgo\m_dispatcher.dll+00018036 Mfunction_handle::dispatch+00000420
[ 70] 0x0000000017042aa1                            bin\win64\pgo\m_lxe.dll+00404129
[ 71] 0x00000000170c8e25                            bin\win64\pgo\m_lxe.dll+00953893 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00115205
[ 72] 0x00000000170ca808                            bin\win64\pgo\m_lxe.dll+00960520 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00121832
[ 73] 0x00000000170cb403                            bin\win64\pgo\m_lxe.dll+00963587 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00124899
[ 74] 0x00000000170cc836                            bin\win64\pgo\m_lxe.dll+00968758 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130070
[ 75] 0x00000000170cd3f0                            bin\win64\pgo\m_lxe.dll+00971760 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00133072
[ 76] 0x00000000170cc97f                            bin\win64\pgo\m_lxe.dll+00969087 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130399
[ 77] 0x00000000170cca7e                            bin\win64\pgo\m_lxe.dll+00969342 mwboost::serialization::singleton<mwboost::archive::detail::pointer_iserializer<mwboost::archive::binaryTerm_iarchive,ir::IrTree> >::get_instance+00130654
[ 78] 0x000000001704e8a5                            bin\win64\pgo\m_lxe.dll+00452773
[ 79] 0x000000001704b335                            bin\win64\pgo\m_lxe.dll+00439093
[ 80] 0x000000001704a82e                            bin\win64\pgo\m_lxe.dll+00436270
[ 81] 0x0000000017051b9a                            bin\win64\pgo\m_lxe.dll+00465818
[ 82] 0x000000001705133d                            bin\win64\pgo\m_lxe.dll+00463677
[ 83] 0x000000001705144e                            bin\win64\pgo\m_lxe.dll+00463950
[ 84] 0x00000000171f1d9c                            bin\win64\pgo\m_lxe.dll+02170268 mwboost::serialization::singleton<mwboost::archive::detail::pointer_oserializer<mwboost::archive::binary_oarchive,MathWorks::lxe::PreLineExecutionEvent> >::get_instance+00475644
[ 85] 0x00000000171f1d4a                            bin\win64\pgo\m_lxe.dll+02170186 mwboost::serialization::singleton<mwboost::archive::detail::pointer_oserializer<mwboost::archive::binary_oarchive,MathWorks::lxe::PreLineExecutionEvent> >::get_instance+00475562
[ 86] 0x0000000017020647                            bin\win64\pgo\m_lxe.dll+00263751
[ 87] 0x00000000163efdc9                    bin\win64\pgo\m_interpreter.dll+00064969 inEvalCmdWithLocalReturnInDesiredWSAndPublishEvents+00000081
[ 88] 0x00000000fcffd8e8                                  bin\win64\iqm.dll+00448744 iqm::InternalEvalPlugin::inEvalCmdWithLocalReturn+00000152
[ 89] 0x00000000fcffd203                                  bin\win64\iqm.dll+00446979 iqm::InternalEvalPlugin::execute+00000211
[ 90] 0x0000000016094781                                  bin\win64\mcr.dll+00411521 mcrRegisterExternalFunction+00063985
[ 91] 0x00000000fcfed9ea                                  bin\win64\iqm.dll+00383466 iqm::Iqm::setupIqmFcnPtrs+00071994
[ 92] 0x00000000fcfed84c                                  bin\win64\iqm.dll+00383052 iqm::Iqm::setupIqmFcnPtrs+00071580
[ 93] 0x00000000fcfd2490                                  bin\win64\iqm.dll+00271504 iqm::Iqm::deliver+00019456
[ 94] 0x00000000fcfc41b7                                  bin\win64\iqm.dll+00213431 iqm::IqmInputRequestEvent::~IqmInputRequestEvent+00009751
[ 95] 0x00000000fd7d7b23                               bin\win64\mlutil.dll+01866531 cmddistributor::IIPRunNowMessage::deliver+00000035
[ 96] 0x000000001171621b                              bin\win64\libmwms.dll+01270299 foundation::msg_svc::exchange::MessageQueue::deliver+00000171
[ 97] 0x0000000011716bd5                              bin\win64\libmwms.dll+01272789 foundation::msg_svc::exchange::MessageQueue::sendReply+00002005
[ 98] 0x0000000011704c2a                              bin\win64\libmwms.dll+01199146 foundation::msg_svc::exchange::CommunicationErrorEvent::getIsError+00072010
[ 99] 0x000000001170f8fc                              bin\win64\libmwms.dll+01243388 foundation::msg_svc::exchange::CommunicationErrorEvent::getIsError+00116252
[100] 0x000000001170ef87                              bin\win64\libmwms.dll+01240967 foundation::msg_svc::exchange::CommunicationErrorEvent::getIsError+00113831
[101] 0x00000000fd6da058                               bin\win64\mlutil.dll+00827480 cmddistributor::IIP::EventMgrBypass::operator=+00037784
[102] 0x00000000fd6ae6b3                               bin\win64\mlutil.dll+00648883 cmddistributor::DebugLoopEvent::serializeExplicit+00012515
[103] 0x00000000fd6df283                               bin\win64\mlutil.dll+00848515 cmddistributor::IIP::EventMgrBypass::operator=+00058819
[104] 0x00000000fd6ee2ff                               bin\win64\mlutil.dll+00910079 cmddistributor::IIPCancelReply::getStatus+00041103
[105] 0x00000000fd7e881b                               bin\win64\mlutil.dll+01935387 cmddistributor::CmdRequestTraits<cmddistributor::FcnCmdRequest<void> >::operator=+00005067
[106] 0x000000001606dfa3                                  bin\win64\mcr.dll+00253859 mcr::runtime::setInterpreterThreadSingletonToCurrent+00122627
[107] 0x0000000016098888                                  bin\win64\mcr.dll+00428168 mcr_set_enableReadingFromStdin+00008168
[108] 0x0000000016086c8a                                  bin\win64\mcr.dll+00355466 mcrRegisterExternalFunction+00007930
[109] 0x00000000160889bc                                  bin\win64\mcr.dll+00362940 mcrRegisterExternalFunction+00015404
[110] 0x0000000016085bff                                  bin\win64\mcr.dll+00351231 mcrRegisterExternalFunction+00003695
[111] 0x0000000016086c46                                  bin\win64\mcr.dll+00355398 mcrRegisterExternalFunction+00007862
[112] 0x0000000016085ce3                                  bin\win64\mcr.dll+00351459 mcrRegisterExternalFunction+00003923
[113] 0x0000000016088ff6                                  bin\win64\mcr.dll+00364534 mcrRegisterExternalFunction+00016998
[114] 0x00000000fd014f67                                  bin\win64\iqm.dll+00544615 iqm::IqmPlugin<cmddistributor::PackagedTaskIIP>::getIIP+00016215
[115] 0x00000000fd01567f                                  bin\win64\iqm.dll+00546431 iqm::PackagedTaskPlugin::execute+00000911
[116] 0x00000000fd014fed                                  bin\win64\iqm.dll+00544749 iqm::IqmPlugin<cmddistributor::PackagedTaskIIP>::getIIP+00016349
[117] 0x00000000fd0154c3                                  bin\win64\iqm.dll+00545987 iqm::PackagedTaskPlugin::execute+00000467
[118] 0x0000000016094931                                  bin\win64\mcr.dll+00411953 mcrRegisterExternalFunction+00064417
[119] 0x00000000fcfed9ea                                  bin\win64\iqm.dll+00383466 iqm::Iqm::setupIqmFcnPtrs+00071994
[120] 0x00000000fcfed84c                                  bin\win64\iqm.dll+00383052 iqm::Iqm::setupIqmFcnPtrs+00071580
[121] 0x00000000fcfcdf0b                                  bin\win64\iqm.dll+00253707 iqm::Iqm::deliver+00001659
[122] 0x00000000fb5fdd33                          bin\win64\libmwbridge.dll+00056627 ioReadLine+00000595
[123] 0x00000000fb5fdbc3                          bin\win64\libmwbridge.dll+00056259 ioReadLine+00000227
[124] 0x00000000fb60b78b                          bin\win64\libmwbridge.dll+00112523 mnDebugPrompt+00001979
[125] 0x00000000fb60b110                          bin\win64\libmwbridge.dll+00110864 mnDebugPrompt+00000320
```


If I launch Matlab from the MSYS2 console it is working perfectly, it is only when launched by Webots.

**Comment by DavidMansolino** [comment]
Regenerating the matlab protofile at each controller start (this can be done automatically from the `launcher.m`) fixes the problem for all the controller not using the plots.

**Comment by DavidMansolino** [comment]
I understand a bit more the problem now, when we start a matlab controller, 'protofile_matlab_2018b_webots_R2019a_revision_1.m' is generated and 'libController_thunk_pcwin64.dll' is compiled in order to use the libController library, this compilation is done using the msys libraries, then Matlab will load this 'libController_thunk_pcwin64.dll' (which depends on some msys libraries, e.g. `msys-2.0.dll`) I assume one of these libraries is not compatible with the one included in Matlab.

**Comment by DavidMansolino** [comment]
Fixed in  #270
